R/opts.R

```diff
@@ -442,6 +461,98 @@ opt_table_outline <- function(data,
   # Perform input object validation
   stop_if_not_gt(data = data)
 
-  values_vec <- rep(c(style, width, color), 4)
-  tab_options_multi(data, create_option_value_list(table_borders_vec(), values_vec))
+  if (style == "none") {
+    width <- NULL
+    color <- NULL
+  }
+
+  params <-
+    c(style = !is.null(style), width = !is.null(width), color = !is.null(color)) %>%
+    which() %>%
+    names()
+
+  pattern <- paste0("^table\\.border.*?\\.(", paste(params, collapse = "|"), ")")
+
+  # Get vector of `tab_options()` arg names for the table border styles
+  options_vec <- get_tab_options_arg_vec(pattern = pattern)
+
+  if (is.null(width)) width <- NA_character_
+  if (is.null(color)) color <- NA_character_
+
+  values_vec <-
+    dplyr::case_when(
+      grepl("^table\\.border.*?\\.style$", options_vec) ~ as.character(style),
+      grepl("^table\\.border.*?\\.width$", options_vec) ~ as.character(width),
+      grepl("^table\\.border.*?\\.color$", options_vec) ~ as.character(color)
+    )
+
+  option_value_list <- create_option_value_list(options_vec, values_vec)
+  tab_options_multi(data, option_value_list)
```

Comment:
See what happens when e.g. `color = NA` is explicitly passed; if this works, then add unit test

Reply:
This currently does not work with `NA`:

```
exibble %>% gt() %>% opt_table_outline(color = NA)
```
```
Error: must be a character vector, not a logical vector
Run `rlang::last_error()` to see where the error occurred.
```

```
exibble %>% gt() %>% opt_table_outline(color = NA_character_)
```
```
Error in (function (option, var_name, type)  : 
  Assertion on 'table_border_top_color' failed: Contains missing values (element 1).
```